File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/book_renewal_model/BookRenewalDAO.java
package com.nintersoft.bibliotecaufabc.book_renewal_model;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
@SuppressWarnings("UnusedReturnValue")
public interface BookRenewalDAO {

    @Query("SELECT * FROM renewal_items")
    List<BookRenewalProperties> getAll();

    @Insert
    long insert(BookRenewalProperties bookRenewalProperties);

    @Delete
    int remove(BookRenewalProperties bookRenewalProperties);

    @Query("DELETE FROM renewal_items")
    void removeAll();
}

File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/model/renewal/BookRenewalDAO.kt
package com.nintersoft.bibliotecaufabc.model.renewal

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface BookRenewalDAO {
    @Query("SELECT * FROM ${BookRenewalContract.TABLE_NAME}")
    fun getAll() : LiveData<List<BookRenewal>>

    @Insert
    fun insert(bookRenewal: BookRenewal) : Long

    @Delete
    fun remove(bookRenewal: BookRenewal) : Int

    @Query("DELETE FROM ${BookRenewalContract.TABLE_NAME}")
    fun removeAll()
}